packages/Webkul/Shop/src/Http/Controllers/API/WishlistController.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+namespace Webkul\Shop\Http\Controllers\API;
+
+use Cart;
+use Illuminate\Http\Resources\Json\JsonResource;
+use Webkul\Customer\Repositories\WishlistRepository;
+use Webkul\Product\Repositories\ProductRepository;
+use Webkul\Shop\Http\Resources\WishlistResource;
+
+class WishlistController extends APIController
+{
+    /**
+     * Create a new controller instance.
+     *
+     * @return void
+     */
+    public function __construct(
+        protected WishlistRepository $wishlistRepository,
+        protected ProductRepository $productRepository
+    ) {
+    }
+
+    /**
+     * Displays the listing resources if the customer having items in wishlist.
+     */
+    public function index(): JsonResource
+    {
+        if (! core()->getConfigData('general.content.shop.wishlist_option')) {
+            abort(404);
+        }
+
+        $this->removeInactiveItems();
+
+        $items = $this->wishlistRepository
+            ->where([
+                'channel_id'  => core()->getCurrentChannel()->id,
+                'customer_id' => auth()->guard('customer')->user()->id,
+            ])
+            ->get();
+
+        return WishlistResource::collection($items);
+    }
+
+    /**
+     * Function to add item to the wishlist.
+     */
+    public function store(): JsonResource
+    {
+        $customer = auth()->guard('customer')->user();
+
+        $product = $this->productRepository->find(request()->input('product_id'));
+
+        if (! $product) {
```

Comment:
Add product remove logic as this is just showing message not removing.

Reply:
Done